src/consensus/validator.rs

```diff
@@ -168,6 +178,10 @@ impl ShardValidator {
             panic!("No proposer set");
         };
 
+        if let Err(err) = self.proposal_store.put_proposal(full_proposal.clone()) {
+            error!("Unable to store proposal {}", err.to_string());
+        };
```

Comment:
I don't think we should do this in `add_proposed_value`, that's going to persist it for every node (and during sync). We only need to remember our own proposals.